lib/Sema/CSApply.cpp

```diff
@@ -2475,25 +2475,34 @@ namespace {
                                          baseTyUnwrapped,
                                          memberName,
                                          defaultMemberLookupOptions);
-          
-          // Lookup didn't find anything, so return
+
+          // Filter out any functions, instance members or enum cases with
+          // associated values.
+          results.filter([](const LookupResultEntry entry, bool isOuter) {
+            if (auto member = entry.getValueDecl()) {
+              if (isa<FuncDecl>(member))
+                return false;
+              if (member->isInstanceMember())
+                return false;
+              if (auto EED = dyn_cast<EnumElementDecl>(member)) {
+                return !EED->hasAssociatedValues();
+              }
+            }
+
+            return true;
+          });
+
           if (results.empty()) {
             return;
           }
           
           if (auto member = results.front().getValueDecl()) {
-            // Lookup returned a member that is an instance member,
-            // so return
-            if (member->isInstanceMember()) {
-              return;
-            }
-            
-            // Return if the member is an enum case w/ assoc values, as we only
-            // care (for now) about cases with no assoc values (like none)
-            if (auto EED = dyn_cast<EnumElementDecl>(member)) {
-              if (EED->hasAssociatedValues()) {
+            // If this is a variable whose type does not match the base type
+            // of the target then ignore it.
+            if (auto VD = dyn_cast<VarDecl>(member)) {
+              auto baseType = DSCE->getType()->lookThroughAllOptionalTypes();
+              if (!VD->getInterfaceType()->isEqual(baseType))
```

Comment:
Why is this one checked here but the other tests done in the filter?

(I was just going to suggest looping over the results and using `continue` instead of `return`, but filtering works too.)

Reply:
No good reason I guess, let me move it into the filter.

Reply:
Done